hashintel/hash
# Add migrations for HASH Graph

## 🌟 What is the purpose of this PR?



This change adds migration scripts for the new type system's DB schemas. 
Link types are not a part of this yet, and some of the aspects of the schemas are best guesses, and are subject to change.


## 🔗 Related links





- [Asana task 1](https://app.asana.com/0/1201095311341924/1202510174412966/f) _(internal)_
- [Asana task 2](https://app.asana.com/0/1202482456346912/1202510174412964/f) _(internal)_

## 🚫 Blocked by




## 🔍 What does this change?




- Moves previous `packages/hash/datastore` package to `packages/graph/migrations`
- Changes DB schema to match new the type system
- Very initial Rust types for the DB 
- Change to `.env` so we do not wipe people's local DBs when trying out the new Graph API

## 📜 Does this require a change to the docs?



- We did change some of the documentation for the `migrations` project, although we need to add more 'getting started' documentation to the new Graph API

## ⚠️ Known issues



## 🐾 Next steps



- Merging https://github.com/hashintel/hash/pull/710
- Considering if the .env variable used by SQLX can be aligned with what we have in the root .env

## 🛡 What tests cover this?



- No automated tests yet

## ❓ How to test this?



1.  Checkout the branch / view the deployment
1.  run `yarn pg:recreate-db` and `yarn db:migrate up` from the migrations directory
1.  Confirm that the DB is created

## 📹 Demo
